web-app/HeimScripts/heatmap/run.R

```diff
@@ -176,3 +189,85 @@ add.subset.label <- function(df,label){
   }
   return(df)
 }
+
+computeDendrogram <- function(zScoreMatrix, distanceMeassure, linkageMethod) {
+    as.dendrogram(hclust(dist(zScoreMatrix, method=distanceMeassure), method=linkageMethod))
+}
```

Comment:
It does matter for the clustering, but not for the calculation of the distance matrix. What is done is that first the distance matrix is calculated, and this distance matrix is then used for clustering with a certain linkage method.
You can break down the command here:
hclust(dist(zScoreMatrix, method=distanceMeassure), method=linkageMethod)
to
calculting distance matrix: distance <- dist(zScoreMatrix, method=distanceMeassure)  -- the result of this will be the same for the three commands 
* colDendrogramEuclideanComplete <- computeDendrogram(t(measurements), 'euclidean', 'complete')
* colDendrogramEuclideanSingle <- computeDendrogram(t(measurements), 'euclidean', 'single')
* colDendrogramEuclideanAverage <- computeDendrogram(t(measurements), 'euclidean', 'average')
cluster the distance matrix: hclust(distance, method=linkageMethod) -- the result of this will differ between the three commands above